infrastructure.md

```diff
@@ -8,4 +8,100 @@ The best description of the full infrastructure is currently given as an [Ansibl
+
+![schematic overview](img/infra-1.png)
+
+## Django website
+
+The website is written in Python using [Django](https://docs.djangoproject.com/en/2.2/).
+The Django server serves up a HTML and javascript front-end, which can interact with the Django server backend via a REST API.
+**JONATHAN: DID WE ADD ANYTHING HERE, OR IS THIS ALL HANDLED BY DJANGO?**
+
+The current Django application is slightly cardiac-specific, and is hosted in the [WebLab](https://github.com/ModellingWebLab/WebLab) repository.
+
+The django back-end can access data from various sources (databases, local files, git repositories, a metadata triple store), and can communicate with the "task queue" to execute experiments via a REST API.
+**JONATHAN: IS THIS API DOCUMENTED SOMEWHERE?**
+
+### Database
+
+PostgreSQL?
+
+### Git repositories
+
+Where are they located? What python module do we use to access them?
+
+### Local file storages
+
+### Triple store
+
+## Running experiments
+
+Experiments are run by calling a `cgi` script, that calls the `fc-runner` (see below).
+
+
+### FC-Runner
+
+The [fc-runner](https://github.com/ModellingWebLab/fc-runner) repository contains a CGI script that can be called by the Django front-end.
+It handles three tasks:
+
+- Determining a protocol interface (i.e. the variable annotations it requires)
+- Scheduling experiments
+- Cancelling scheduled experiments
+
+### Scheduling experiments
+
+Experiments are schedule by placing them into a [Celery](http://docs.celeryproject.org/en/latest/index.html) task queue ([wiki](https://en.wikipedia.org/wiki/Celery_(software)).
+One or multiple workers then take tasks from the queue and execute them.
+This way, the actual running of experiments can be offloaded to a cloud of worker machines.
+
+The workers talk to the queue via a message passing system.
+Instead of talking to each other directly, messaging is handled via a *broker*, for which Web Lab uses [RabbitMQ](https://www.rabbitmq.com/documentation.html) ([wiki](https://en.wikipedia.org/wiki/RabbitMQ)).
+See also [Celery docs: Using RabbitMQ](http://docs.celeryproject.org/en/latest/getting-started/brokers/rabbitmq.html).
+
+Further information is given [here](./infrastructure-cardiac.md).
+
+## Deployment with Vagrant and Ansible
+
+[Vagrant](https://www.vagrantup.com/docs/index.html) ([wiki](https://en.wikipedia.org/wiki/Vagrant_%28software%29)) is a tool for "building and maintaining virtual software development environments".
+For the WebLab, we use Vagrant to create and manage [VirtualBox](https://en.wikipedia.org/wiki/VirtualBox) machines, which are automatically set up for development or production using Ansible.
+Users can connect to running vagrant machines using ``$ vagrant ssh``.
+
+[Ansible](https://docs.ansible.com/) ([wiki](https://en.wikipedia.org/wiki/Ansible_(software))) is a tool to set up development environments, or deploy applications to some production environment.
```

Comment:
Actually it's primarily for doing 'infrastructure as code' and setting up production environments. But we have a slightly tweaked version that can set up a completely local instance of full Web Lab. The primary difference is it only serves locally via http, not https.